python/ray/train/dataset_spec.py

```diff
@@ -0,0 +1,94 @@
+from dataclasses import dataclass
```

Comment:
If this file is internal, can we move it to `train/impl/dataset_spec.py`? Let's try to have a convention for impl vs public api files.

Reply:
Moved to impl!